lib/ansible/modules/cloud/amazon/code_pipeline.py

```diff
@@ -0,0 +1,293 @@
+                         exception=traceback.format_exc())
+
+
+def delete_pipeline(client, name, module):
+    try:
+        resp = client.delete_pipeline(name=name)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable delete pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to delete pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def describe_pipeline(client, name, version, module):
+    pipeline = {}
+    try:
+        if isinstance(version, int) is int:
+            pipeline = client.get_pipeline(name=name, version=version)
+            return pipeline
+        else:
+            pipeline = client.get_pipeline(name=name)
+            return pipeline
+    except botocore.exceptions.ClientError as e:
+        return pipeline
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Error when calling client.get_pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True, type='str'),
+        role_arn=dict(required=True, type='str'),
+        artifact_store=dict(required=True, type='dict'),
+        stages=dict(required=True, type='list'),
+        version=dict(required=False, type='int'),
+        state=dict(choices=['present', 'absent'], default='present')
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    client_conn = boto3_conn(module, conn_type='client', resource='codepipeline', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    state = module.params.get('state')
+    changed = False
+
+    # Determine if the CodePipeline exists
+    found_code_pipeline = describe_pipeline(client=client_conn, name=module.params['name'], version=module.params['version'], module=module)
+    pipeline_result = {}
+
+    if state == 'present':
+        if 'pipeline' in found_code_pipeline:
+            pipeline_dict = found_code_pipeline['pipeline']
+            # Update dictionary with provided module params:
+            pipeline_dict['roleArn'] = module.params['role_arn']
+            pipeline_dict['artifactStore'] = module.params['artifact_store']
+            pipeline_dict['stages'] = module.params['stages']
+            if isinstance(module.params['version'], int):
+                pipeline_dict['version'] = module.params['version']
+
+            pipeline_result = update_pipeline(client=client_conn, pipeline_dict=pipeline_dict, module=module)
```

Comment:
It looks like there is no code path when state is present for changed to be False. Should the above params be compared with the attributes of the found_code_pipeline and only update if they are different instead? Since updating bumps the version number, there could be a `force_update` module option if that's needed.